httpx/_client.py

```diff
@@ -619,6 +639,7 @@ def _transport_for_url(self, url: URL) -> httpcore.SyncHTTPTransport:
 
         return self._transport
 
+    @check_not_closed
```

Comment:
I guess we should instead put the check around `def send()` rather than `def request()`.

Reply:
Rather than the decorator, I'd prefer the plainer style of starting the method with...

```python
if self._is_closed:
   raise RuntimeError("Cannot send requests on a closed client instance.")
```

Reply:
> I guess we should instead put the check around def send() rather than def request().

Done

> Rather than the decorator, I'd prefer the plainer style of starting the method with...

Done, but I used `self.is_closed` in the condition